mesonbuild/mesonlib.py

```diff
@@ -378,7 +393,7 @@ def _get_machine_is_cross(env, is_cross):
         env.machines[MachineChoice.HOST].is_haiku()
 
     """
-    for_machine = MachineChoice.HOST if is_cross else MachineChoice.BUILD
+    for_machine = MachineChoice.is_native(not is_cross)
```

Comment:
OK yes this is another call site, but in #4010 I remove this and all the `for_*` uses.